lib/node_modules/@stdlib/math/base/special/log10/src/main.c

```diff
@@ -160,56 +69,81 @@ static double klog( const double x ) {
 * // returns ~0.602
 */
 double stdlib_base_log10( const double x ) {
-	int32_t ihx;
-	int32_t ilx;
+	double valLo;
+	double valHi;
 	uint32_t hx;
 	uint32_t lx;
+	double hfsq;
+	double hi;
 	int32_t i;
 	int32_t k;
-	double xc;
-	double hi;
 	double lo;
+	double xc;
+	double y2;
 	double f;
+	double R;
+	double w;
 	double y;
-	double z;
 
+	if ( stdlib_base_is_nan( x ) || x < 0.0 ) {
+		return 0.0 / 0.0; // NaN
+	}
 	xc = x;
-	stdlib_base_float64_to_words( x, &hx, &lx );
-	ihx = (int32_t)hx;
-	ilx = (int32_t)lx;
+	stdlib_base_float64_to_words( xc, &hx, &lx );
 	k = 0;
-
-	// Case: 0 < x < 2**-1022
-	if ( ihx < HIGH_MIN_NORMAL_EXP ) {
-		if ( ( ( ihx & STDLIB_CONSTANT_FLOAT64_HIGH_WORD_ABS_MASK ) | ilx ) == 0 ) {
+	if ( hx < HIGH_MIN_NORMAL_EXP ) {
+		// Case: x < 2**-1022
+		if ( ( ( hx & STDLIB_CONSTANT_FLOAT64_HIGH_WORD_ABS_MASK ) | lx ) == 0 ) {
```

Comment:
@gunjjoshi @Planeshifter Correct me if I'm wrong, but `hx` is `uint32_t` here. Bitwise ops are defined for signed integers. Meaning, we should still be using `ihx` and `ilx` here and below.

Reply:
I see that we do this in `log2` (https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/log2/src/main.c#L75), but I am not sure that is correct or works as intended.

Reply:
In https://github.com/stdlib-js/stdlib/blob/d3215eb50a3837122325f8db0415797f81d46128/lib/node_modules/%40stdlib/math/base/special/acosf/src/main.c#L104, we do the explicit cast to `int32_t`. IIRC, the failure to explicitly cast words to signed integers resulted in test failures in other packages. 

**Update**: after digging around a bit, looks like all bitwise operators are defined for unsigned integers. The issue arises when mixing signed and unsigned integer values. In that instance, unsigned integers will cast to signed integers which can cause unexpected behavior. My sense is that wherever FreeBSD uses signed integers, we should also use signed integers. In that case, we should use `ihx` (see https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_log10.c?view=markup).

Reply:
Yes @kgryte, the process of casting `hx` to a signed integer after these operations seems correct. I'll try to reiterate over this implementation once again, keeping this in mind.